modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -1,39 +1,35 @@
 require 'resolv'
 require 'open3'
+require 'dns_common/errors'
+require 'dns_common/record'
 
 module Proxy::Dns::Dnscmd
   class Record < ::Proxy::Dns::Record
     include Proxy::Log
     include Proxy::Util
-    attr_reader :resolver
 
-    def self.record(attrs = {})
-      new(attrs.merge(:server => ::Proxy::Dns::Dnscmd::Plugin.settings.dns_server,
-                      :ttl => ::Proxy::Dns::Plugin.settings.dns_ttl))
+    def initialize(server = nil, ttl = nil)
+      super(server || ::Proxy::Dns::Dnscmd::Plugin.settings.dns_server, ttl || ::Proxy::Dns::Plugin.settings.dns_ttl)
```

Comment:
Nitpick, can you split this into a few lines?  Probably `server ||= ::Proxy....settings..` and same for ttl, it'll be easier to read.